Makefile

```diff
@@ -9,6 +9,10 @@ all: build/CMakeCache.txt
 .DEFAULT: build/CMakeCache.txt
 	ninja -C ${build} $@
 
+# empty default to ensure build dir is created when make called with arguments
+%:
+	#
```

Comment:
Please add something to CI that doesn't work without this fix. If I understand correctly it could be a simple thing like `make test TESTS=`

Reply:
I don't want to build a dependency circle with our tests testing the call of themselves. That does not make sense

Reply:
I don't know what that means. I'm suggesting to make sure we call make targets in a way that reproduces the issue in CircleCI or GHA

Reply:
Test added with https://github.com/os-autoinst/os-autoinst/pull/1884/commits/c2a17030e5f05198f062810ae52ee68cda3fb438 which is using a local git clone to a temporary directory to not mess up the original source directory